scripts/track_upstream_commits/track_upstream_commits.py

```diff
@@ -0,0 +1,184 @@
+    )
+
+    # Remove duplicate commits from each crate
+    for crate in crates_commits:
+        crates_commits[crate] = [commit for commit in crates_commits[crate] if commit not in duplicate_commits]
+
+    # Print the name of the crate, number of commits contained, and the commits
+    for crate, commits in crates_commits.items():
+        if commits:
+            print(f"\n\n## {crate} ({len(commits)})")
+            for commit in reversed(commits):
+                print(f"- {commit}")
+
+    # Print the duplicate commits
+    print(f"\n\n## Cross-crate commits ({len(duplicate_commits)})")
+    for commit in reversed(list(duplicate_commits)):
+        print(f"- {commit}")
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Track upstream commits for specified crates.')
+    parser.add_argument('--since', required=True, help='Start commit hash for git log (e.g., "bb778828e36d53a7d91a27e55109f2f45621badc").')
+    parser.add_argument('--until', required=True, help='End commit hash for git log (e.g., "3ada97c109cc7ae1b451cb384a1f2cfae49c8d3e"), it is included in the results.')
+    parser.add_argument('--crates', nargs='+', help='List of crates to track (e.g., "iota-core iota-node").')
+    parser.add_argument('--codeowner', help='code owner of the crates (e.g., "node)')
+    parser.add_argument('--repo-url', default="git@github.com:MystenLabs/sui.git", help="The URL to the repository. Can also be a local folder.")
+    parser.add_argument('--repo-tag', default=None, help="The tag to checkout in the repository.")
+    parser.add_argument('--version', default=None, help="The semantic version to filter overwrites/patches if not found in the repo-tag.")
+    parser.add_argument('--target-folder', default="result", help="The path to the target folder.")
+    parser.add_argument('--clone-source', action='store_true', help="Clone the upstream repository.")
+    parser.add_argument('--compare-source-folder', help="The path to the source folder for comparison.")
+
+    args = parser.parse_args()
+    target_folder = args.target_folder
+    target_folder = os.path.abspath(os.path.expanduser(target_folder))
+    
+    # Check if clone_source is true and repo_tag is not specified
+    if args.clone_source and not args.repo_tag:
+        parser.error("--repo-tag must be specified if --clone-source is true")
+
+    # get current root folder
+    source_folder = os.path.abspath(os.path.join(os.getcwd(), "..", ".."))
+    if args.compare_source_folder:
+        source_folder = os.path.abspath(args.compare_source_folder)
+
+    # Check if crates or code owner is specified
+    if not args.crates and not args.codeowner:
+        print("No crates or code owner specified.")
+        exit(1)
+
+    crates = []
+    # Replace 'iota' with 'sui', and add the prefix 'crates/'
+    if args.crates:
+        crates = [f"crates/{crate.replace('iota', 'sui')}" for crate in args.crates]
```

Comment:
I think we also renamed some crates, so that won't work 100%. Maybe you need some mapping table?